cmd/plugin/builder/inventory/inventory_plugin.go

```diff
@@ -0,0 +1,165 @@
+// and publishing the inventory database as OCI image on the remote repository
+func (ipuo *InventoryPluginUpdateOptions) PluginInsert() error {
+	pluginInsertFunc := func(dbFile string, entry *plugininventory.PluginInventoryEntry) error {
+		db := plugininventory.NewSQLiteInventory(dbFile, "")
+		err := db.InsertPlugin(entry)
+		if err != nil {
+			return errors.Wrapf(err, "error while inserting plugin '%s_%s'", entry.Name, entry.Target)
+		}
+		return nil
+	}
+	return ipuo.genericInventoryUpdater(pluginInsertFunc)
+}
+
+// UpdatePluginActivationState updates plugin entry in the inventory database by downloading the
+// database from the repository, updating it locally and publishing the inventory database
+// as OCI image on the remote repository
+func (ipuo *InventoryPluginUpdateOptions) UpdatePluginActivationState() error {
+	activateDeactivateFunc := func(dbFile string, entry *plugininventory.PluginInventoryEntry) error {
+		db := plugininventory.NewSQLiteInventory(dbFile, "")
+		err := db.UpdatePluginActivationState(entry)
+		if err != nil {
+			return errors.Wrapf(err, "error while updating plugin '%s_%s'", entry.Name, entry.Target)
+		}
+		return nil
+	}
+	return ipuo.genericInventoryUpdater(activateDeactivateFunc)
+}
+
+func (ipuo *InventoryPluginUpdateOptions) genericInventoryUpdater(inventoryUpdater func(string, *plugininventory.PluginInventoryEntry) error) error {
+	// create plugin inventory database image path
+	pluginInventoryDBImage := fmt.Sprintf("%s/%s:%s", ipuo.Repository, helpers.PluginInventoryDBImageName, ipuo.InventoryImageTag)
+
+	dir, err := os.MkdirTemp("", "")
+	if err != nil {
+		return errors.Wrap(err, "unable to create temporary directory")
+	}
+
+	log.Infof("Pulling plugin inventory database from: %q", pluginInventoryDBImage)
+	err = ipuo.ImgpkgOptions.PullImage(pluginInventoryDBImage, dir)
+	if err != nil {
+		return errors.Wrapf(err, "error while pulling database from the image: %q", pluginInventoryDBImage)
+	}
+
+	dbFile := filepath.Join(dir, plugininventory.SQliteDBFileName)
+	pluginInventoryEntries, err := ipuo.getPluginInventoryEntries()
+	if err != nil {
+		return errors.Wrap(err, "error while updating plugin inventory database")
+	}
+
+	for i := range pluginInventoryEntries {
+		err := inventoryUpdater(dbFile, pluginInventoryEntries[i])
+		if err != nil {
+			return err
+		}
+	}
+
+	// Publish the database to the remote repository
+	log.Info("Publishing plugin inventory database")
+	err = ipuo.ImgpkgOptions.PushImage(pluginInventoryDBImage, dbFile)
+	if err != nil {
+		return errors.Wrapf(err, "error while publishing inventory database to the repository as image: %q", pluginInventoryDBImage)
+	}
+	log.Infof("Successfully published plugin inventory database at: %q", pluginInventoryDBImage)
+
+	return nil
+}
+
+func (ipuo *InventoryPluginUpdateOptions) getPluginInventoryEntries() ([]*plugininventory.PluginInventoryEntry, error) {
```

Comment:
nit: Should we name this method to reflect that we prepare the plugin inventory entries from plugin manifest file.

Reply:
Ack. Updated the function name to `preparePluginInventoryEntriesFromManifest`